applications/FluidDynamicsHydraulicsApplication/python_scripts/apply_hydraulic_inlet_process.py

```diff
@@ -0,0 +1,181 @@
+        current_time = self.inlet_model_part.ProcessInfo[KratosMultiphysics.TIME]
+        if self.interval.IsInInterval(current_time):
+            self.step_is_active = True
+            if self.discharge_value_is_function:
+
+                if self.aux_function.DependsOnSpace() == False:  # depends on time only
+                    self.inlet_discharge = self.aux_function.CallFunction(0.0, 0.0, 0.0, current_time, 0.0, 0.0, 0.0)
+                else:  # Only time-dependent hydrograms are implemented.
+                    err_msg = "There is no space-dependent function implemented "
+                    raise ValueError(err_msg)
+
+            elif self.discharge_value_is_table:
+                self.inlet_discharge = self.table.GetValue(current_time)
+
+        #Calculate the initial water depth guess.
+        if not self.initial_water_depth:
+            self.initial_water_depth = KratosFluidHydraulics.HydraulicFluidAuxiliaryUtilities.InitialWaterDepth(
+                self.inlet_model_part)
+
+        # Determine the critical water depth based on the given inlet discharge value and the shape of the inlet model part. It is an iterative process based on the bisection method.
+
+        # Define an interval within which the critical water depth belongs.
+        # TODO: There is a more optimal way to define this interval in order to avoid iterations.
+        water_depth_a = 1e-6
+        water_depth_b = max([node.Z for node in self.inlet_model_part.Nodes])
+        # The froude number for supercritical(small water depth) flows has a big value while for subcritical(High water depth) flows the froude number is small less than one.
+        froude_number_a = 100000
+        froude_number_c = 0.0
+
+        #  Bisection method.
+        iter_count = 0
+        while (abs(froude_number_c - 1) > self.bisection_tolerance) and iter_count < self.maximum_iterations:
+            water_depth_c = 0.5*(water_depth_a+water_depth_b)
+            froude_number_c = self.CalculateFroudeNumber(water_depth_c)
+            if (froude_number_c-1)*(froude_number_a-1)<0:
+                water_depth_b=water_depth_c
+            else:
+                water_depth_a=water_depth_c
+                froude_number_a = froude_number_c
+            iter_count +=1
+        if iter_count == self.maximum_iterations:
+            KratosMultiphysics.Logger.PrintWarning("::[KratosFluidHydraulics]::", "Maximum iteractions for calculating critical water depth have been exceeded.")
+
+        # Obtain the corresponding velocity according to the wetted area and the inlet discharge value.
+        inlet_velocity = self.inlet_discharge/self.wetted_area
+
+        # Assing the inlet velocity to inlet nodes and fix it.
+        # TODO: Create a variable in kratos por Inlet normal, now it the variable uses for that is DISPLACEMENT.
+        KratosMultiphysics.NormalCalculationUtils().CalculateOnSimplexNonHistorical(self.inlet_model_part,self.domain_size,KratosMultiphysics.DISPLACEMENT)
```

Comment:
We should use a different variable.

Reply:
working on, we are goint to move a variable into FluidDynamicsApplication